lib/python/satos_payload_sdk/antaris_api_can.py

```diff
@@ -0,0 +1,152 @@
+
+# This file assumes that, env.json file is present at /opt/antaris/app
+# location. The sample file is checked-in in conf directory
+import can
+import threading
+import time
+import json
+
+g_JSON_Key_IO_Access = "IO_Access"
+g_JSON_Key_CAN = "CAN"
+g_JSON_Key_Device_Count = "CAN_PORT_COUNT"
+g_JSON_Key_CAN_Device_Path = "CAN_Bus_Path_"
+
+# Read config info
+jsonfile = open('/opt/antaris/app/config.json', 'r')
+
+g_GPIO_ERROR = -1
+
+# threading lock
+threadLock = -1
+data_array = []
+
+# returns JSON object as a dictionary
+jsfile_data = json.load(jsonfile)
+
+class CAN:
+    def __init__(self, port_count, can_dev):
+        self.can_port_count = port_count
+        self.can_dev = can_dev
+
+def api_pa_pc_get_can_dev():
+    g_total_can_port = jsfile_data[g_JSON_Key_IO_Access][g_JSON_Key_CAN][g_JSON_Key_Device_Count]
+    can_dev = []
+
+    i = 0
+    for i in range(int(g_total_can_port)):
+        key = g_JSON_Key_CAN_Device_Path+str(i)
+        element = jsfile_data[g_JSON_Key_IO_Access][g_JSON_Key_CAN][key]
+        can_dev.append(element)
+
+    canObj = CAN(g_total_can_port, can_dev)
+    return canObj
+
+def api_pa_pc_receive_can_message(channel, data_array, lock):
+    try:
+        # Initialize a CAN bus interface
+        bus = can.interface.Bus(channel=channel, bustype='socketcan')
+
+        # Continuously receive CAN messages
+        while True:
+            # Receive a CAN message
+            message = bus.recv(timeout=1)
+            if message is not None:
+                with lock:
+                    data_array.append(message)  # Append the received message to the data array
+
+    except can.CanError as e:
+        print("Error receiving message:", e)
+        return g_GPIO_ERROR
+    
+    finally:
+        # Clean up and close the bus
+        bus.shutdown()
+
+def api_pa_pc_read_can_data():
+    with threadLock:
+        if data_array:
+            data = data_array.pop(0)
+            return data 
+        else:
+            return g_GPIO_ERROR
+
+def api_pa_pc_start_can_receiver_thread(channel):
+    # Create shared data array and lock
+    threadLock = threading.Lock()
+    receive_thread = threading.Thread(target=api_pa_pc_receive_can_message, args=(channel, data_array, threadLock))
+    receive_thread.daemon = True
+    receive_thread.start()
+
+def api_pa_pc_send_can_message(channel, arbitration_id, data):
+    try:
+        # Initialize a CAN bus interface
+        bus = can.interface.Bus(channel=channel, bustype='socketcan')
+
+        # Send a CAN message
+        message = can.Message(arbitration_id=arbitration_id, data=data, is_extended_id=False)
+        bus.send(message)
+        return 0
+
+    except can.CanError as e:
+        print("Error sending message:", e)
+        return g_GPIO_ERROR
+    
+    finally:
+        # Clean up and close the bus
+        bus.shutdown()
+
+def api_pa_pc_get_can_message_received_count():
+    return len(data_array)
+
+if __name__ == "__main__":
```

Comment:
I don't think this test function belongs here. Please move it to an actual test/example file.

Reply:
Or if the example_app.py already encompasses it, then you can just delete this

Reply:
I had added this function for standalone testing. But now I removed